instantsearch/src/main/java/com/algolia/instantsearch/model/AlgoliaSearcherListener.java

```diff
@@ -0,0 +1,16 @@
+package com.algolia.instantsearch.model;
+
+import android.support.annotation.NonNull;
+
+import com.algolia.instantsearch.helpers.Searcher;
+
+public interface AlgoliaSearcherListener {
```

Comment:
nit: You can some javadoc here explaining what is this interface for:
- for widgets who need a reference to a Searcher
- Mainly used for "input" widgets that will need to modify the state of the searcher